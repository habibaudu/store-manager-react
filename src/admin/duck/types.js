const LOAD_ALL_PRODUCT = 'LOAD_PRODUCTS';
const ADD_PRODUCT = 'ADD_PRODUCT';
const SET_ADD_STATE = 'SET_ADD_STATE';
const SET_ADD_ERROR = 'SET_ADD_ERROR';
const PRODUCT_SUCESSFULLY_ADDED = 'PRODUCT_SUCESSFULLY_ADDED';
const ADDING_PRODUCT = 'ADDING_PRODUCT';
const ADD_ERROR = 'ADD_ERROR';

export default { LOAD_ALL_PRODUCT, ADD_PRODUCT,PRODUCT_SUCESSFULLY_ADDED,ADDING_PRODUCT,ADD_ERROR,SET_ADD_STATE,SET_ADD_ERROR};
