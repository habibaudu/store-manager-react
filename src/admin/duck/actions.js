import types from './types';

const setAllProducts = products => ({
  type: types.LOAD_ALL_PRODUCT,
  payload: products,
});

const addProducts = productadded => ({
  type: types.ADD_PRODUCT,
  productadded,
});

const addProductStatus = status => ({
  type: types.SET_ADD_STATE,
  status,
});

const addProductError = errormessage => ({
  type: types.SET_ADD_ERROR,
  errormessage,
});

const setDeleteProduct = id => ({
  type: types.DELETE_PRODUCT,
  id,
});

const setSignupState = signupState => ({
  type: types.SET_SIGNUP_STATE,
  signupState,
});
const setSignupError = errorMessage => ({
  type: types.SET_SIGNUP_ERROR,
  errorMessage,
});

export default {
  setAllProducts,
  addProducts,
  addProductStatus,
  addProductError,
  setDeleteProduct,
  setSignupState,
  setSignupError,
};
