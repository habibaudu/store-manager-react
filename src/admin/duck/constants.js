export default {
  PRODUCT_SUCESSFULLY_ADDED: 'PRODUCT_SUCESSFULLY_ADDED',
  ADDING_PRODUCT: 'ADDING_PRODUCT',
  ADD_ERROR: 'ADD_ERROR',
  SIGNING_UP: 'SIGNING_UP',
  SIGNUP_SUCCESS: 'SIGNUP_SUCCESS',
  SIGNUP_ERROR: 'SIGNUP_ERROR',
};
