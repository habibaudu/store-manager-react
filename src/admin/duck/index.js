export { default as adminReducer } from './reducers';
export { default as constants } from './constants';
export { fetchAllProducts, deleteProduct, createAttendant } from './operations';
export { addProduct } from './operations';
export { default as actions } from './actions';
export { default as types } from './types';
