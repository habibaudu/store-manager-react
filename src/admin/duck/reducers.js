import types from './types';

const adminReducer = (state = {}, action) => {
  switch (action.type) {
    case types.LOAD_ALL_PRODUCT: {
      const { payload } = action;
      return { ...state, payload };
    }
    case types.ADD_PRODUCT: {
      const { productadded } = action;
      return { ...state, productadded };
    }
    case types.SET_ADD_STATE: {
      const { status } = action;
      return { ...state, status };
    }
    case types.SET_ADD_ERROR: {
      const { errormessage } = action;
      return { ...state, errormessage };
    }
    case types.DELETE_PRODUCT: {
      const { id } = action;
      const newPayload = state.payload.filter(product => product.id !== id);
      return { ...state, payload: newPayload };
    }
    case types.SET_SIGNUP_STATE: {
      const { signupState, data } = action;
      return {
        ...state,
        signupState,
        data,
      };
    }
    case types.SET_SIGNUP_ERROR: {
      const { errorMessage } = action;
      return {
        ...state,
        errorMessage,
      };
    }
    default:
      return state;
  }
};

export default adminReducer;
