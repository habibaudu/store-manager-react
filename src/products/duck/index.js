export { default as productReducer } from './reducers';
export { default as constants } from './constants';
export { fetchAllProducts, searchProduct } from './operations';
export { default as actions } from './actions';
export { default as types } from './types';
