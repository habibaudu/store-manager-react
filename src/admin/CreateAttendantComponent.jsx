import React, { Fragment } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { constants } from './duck';
import { createAttendant } from './duck';
import { connect } from 'react-redux';

const CreateAttendantComponent = ({
  signupUser,
  signupState,
  errorMessage,
}) => {
  const onFormSubmit = e => {
    e.preventDefault();
    const email = e.target.elements.email.value.trim();
    const username = e.target.elements.username.value.trim();
    const password = e.target.elements.password.value.trim();
    const password2 = e.target.elements.password2.value.trim();
    const Role = e.target.elements.role.value.trim();
    if (password !== password2) {
      alert('passwords do not match');
      return;
    }
    signupUser(email, password, username, Role);
  };

  signUserOut = () => {
    localStorage.removeItem('token');
    return <Redirect to="/" />;
  };

  return (
    <div>
      <Fragment>
        <nav>
          <div className="logo">
            <h4>Store Manager</h4>
          </div>
          <ul className="nav-links">
            <li>
              {' '}
              <Link to="/admin">Admin</Link>
            </li>
            <li>
              <Link
                to="/"
                onClick={() => {
                  signUserOut();
                }}
              >
                Logout
              </Link>
            </li>
          </ul>
          <div className="burger">
            <div className="line1" />
            <div className="line2" />
            <div className="line3" />
          </div>
        </nav>

        <div className="formbox">
          {signupState === constants.SIGNING_UP &&
            'Creating Attendant please wait....'}
          {signupState === constants.SIGNUP_SUCCESS &&
            'Attendant created sucessfully'}
          {errorMessage === constants.SIGNUP_ERROR &&
            'An error occured while creating attendant'}
          <div className="log_head">
            <h1>Create User</h1>
          </div>

          <form onSubmit={onFormSubmit}>
            <p>User name</p>
            <input
              type="text"
              name="username"
              placeholder="Enter Username"
              id="username"
            />
            <p>Email</p>
            <input
              type="email"
              name="email"
              placeholder="Enter Email"
              id="email"
            />
            <p>Password</p>
            <input
              type="password"
              name="password"
              placeholder="Enter password"
              id="password"
            />
            <p>confirm password</p>
            <input
              type="password"
              name="password2"
              placeholder="confirm password"
              id="password2"
            />
            <p>Role</p>
            <br />
            <select name="role" id="Role">
              <option value="ADMIN">ADMIN</option>
              <option value="USER">USER</option>
            </select>
            <br />

            <input type="submit" name="" value="submit" />
          </form>
          <div className="form_footer">
            <p>Store Manager copyright &copy 2018 </p>
          </div>
        </div>
      </Fragment>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    signupState: state.adminReducer.signupState,
    errormessage: state.adminReducer.errormessage,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    signupUser: (email, password, username, Role) =>
      dispatch(createAttendant(email, password, username, Role)),
  };
};
const CreateAttendantContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(CreateAttendantComponent);

export { CreateAttendantContainer };
