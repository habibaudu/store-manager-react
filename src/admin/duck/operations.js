import axios from 'axios';
import { toast } from 'react-toastify';
import actions from './actions';
import constants from './constants';

const API_URL = process.env.API_URL;
const {
  setAllProducts,
  addProducts,
  addProductStatus,
  addProductError,
  setDeleteProduct,
  setSignupError,
  setSignupState,
} = actions;

const url = `${API_URL}products`;

const fetchAllProducts = () => dispatch => {
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };
  return axios
    .get('https://store-manager2.herokuapp.com/api/v1/products', defaultOptions)
    .then(({ data }) => {
      const img = data.imgArr;
      const data2 = data.rows;
      let count = 0;
      const products = [];
      data2.forEach(product => {
        const {
          id,
          productname,
          price,
          quantity,
          minimum,
          description,
          modified_date,
          created_date,
        } = product;

        products.push({
          item: img[count],
          id,
          productname,
          price,
          quantity,
          minimum,
          description,
          modified_date,
          created_date,
        });

        count += 1;
      });
      dispatch(setAllProducts(products));
    })
    .catch(({ response }) => {
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export const addProduct = (
  productname,
  price,
  minimum,
  quantity,
  images,
  description,
) => dispatch => {
  dispatch(addProductStatus(constants.ADDING_PRODUCT));
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };

  return axios
    .post(
      'https://store-manager2.herokuapp.com/api/v1/products',
      { productname, minimum, description, images, price, quantity },
      defaultOptions,
    )
    .then(({ data }) => {
      dispatch(addProductStatus(constants.PRODUCT_SUCESSFULLY_ADDED));
      dispatch(addProducts(data));
    })
    .catch(({ response }) => {
      toast.error(response, {
        hideProgressBar: true,
      });
      dispatch(addProductError(constants.ADD_ERROR));
    });
};

export const deleteProduct = id => dispatch => {
  const deleteUrl = `https://store-manager2.herokuapp.com/api/v1/products/${id}`;
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };

  return axios
    .delete(deleteUrl, defaultOptions)
    .then(({ data }) => {
      dispatch(setDeleteProduct(data));
    })
    .catch(({ response }) => {
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export const createAttendant = (
  email,
  password,
  username,
  Role,
) => dispatch => {
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };
  dispatch(setSignupState(constants.SIGNING_UP));
  dispatch(setSignupError(''));
  return axios
    .post(
      `https://store-manager2.herokuapp.com/api/v1/auth/signup`,
      {
        email,
        password,
        username,
        Role,
      },
      defaultOptions,
    )
    .then(({ data }) => {
      dispatch(setSignupState(constants.SIGNUP_SUCCESS));
    })
    .catch(({ response }) => {
      dispatch(setSignupState(constants.SIGNUP_ERROR));
      dispatch(setSignupError(response));
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export { fetchAllProducts };
