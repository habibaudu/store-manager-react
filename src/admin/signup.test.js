import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Redirect, MemoryRouter } from 'react-router-dom';
import axios from 'axios';
import { CreateAttendantContainer } from './CreateAttendantComponent';
import { actions, types, constants, adminReducer } from './duck';

jest.mock('axios');

const { setSignupState, setSignupError } = actions;

describe('SIGNUP TEST SUITE', () => {
  describe('Signup Component', () => {
    it('should render the Signup Page', () => {
      const component = shallow(<CreateAttendantContainer />);
      expect(component).toMatchSnapshot();
    });

    it('it should not submit the form if any input field is empty', () => {
      const signupUser = jest.fn();
      const props = {
        signupUser,
        signupState: '',
        errorMessage: 'Email password username',
      };
      const component = shallow(<CreateAttendantContainer {...props} />);
      component.find('form').simulate('submit', {
        preventDefault: () => {},
        target: {
          elements: {
            username: { value: 'johndoe' },
            email: { value: 'johndoe@joe.com' },
            password: { value: 'johndoe88' },
          },
        },
      });
      expect(signupUser).not.toHaveBeenCalled();
    });
  });
});
