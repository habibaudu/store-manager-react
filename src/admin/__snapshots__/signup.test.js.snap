// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGNUP TEST SUITE Signup Component should render the Signup Page 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
